Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/AddDocStrucTypeDialog.java

```diff
@@ -418,6 +419,19 @@ public void prepare() {
         prepareSelectPageOnAddNodeItems();
     }
 
+    private void checkSelectedLogicalNode() {
+        //If a view is selected in logical tree then the 'selectedLogicalNode' will be set to the parent of this view
+        TreeNode selectedLogicalNode = dataEditor.getStructurePanel().getSelectedLogicalNode();
+        if (Objects.nonNull(selectedLogicalNode) && selectedLogicalNode.getData() instanceof StructureTreeNode) {
+            StructureTreeNode structureTreeNode = (StructureTreeNode) selectedLogicalNode.getData();
+            if (structureTreeNode.getDataObject() instanceof View) {
+                if (Objects.nonNull(selectedLogicalNode.getParent())) {
+                    dataEditor.getStructurePanel().setSelectedLogicalNode(selectedLogicalNode.getParent());
```

Comment:
I am not sure about this line: I understand that the selected logical node must be set to the parent structure when a page is selected, but what happens when the user then clicks `Cancel`? Then the `selectedLogicalNode` in the `StructurePanel` is still set to the parent structure, even though that is not visible in the tree, where the page still seems to be selected.